### PR TITLE
llvm-reduce: Don't print verifier failed machine functions

### PR DESCRIPTION
**Pull request by arsenm**
This produces far too much terminal output, particularly for the
instruction reduction. Since it doesn't consider the liveness of of
the instructions it's deleting, it produces quite a lot of verifier
errors.

### CODE REVIEW

**Comment by regehr** [comment]
LGTM, unless this wants a test on it

**Review by aeubanks — COMMENTED** [review]
what about passing a `raw_ostream&` to the machine verifier that defaults to `errs()`? and we can pass `nulls()` from llvm-reduce

**Comment by aeubanks** [comment]
or to match the LLVM IR verifier more, a `raw_ostream *` that defaults to `nullptr`

**Review by qcolombet — APPROVED** [review]

File: llvm/lib/CodeGen/MachineVerifier.cpp [hunk 1]
```diff
@@ -94,21 +94,24 @@ using namespace llvm;
...
-          Indexes(Indexes) {}
+                    SlotIndexes *Indexes, raw_ostream *OS)
+        : OS(OS ? OS : &nulls()), Banner(b), LiveVars(LiveVars),
+          LiveInts(LiveInts), LiveStks(LiveStks), Indexes(Indexes) {}
 
     unsigned verify(const MachineFunction &MF);
 
     MachineFunctionAnalysisManager *MFAM = nullptr;
     Pass *const PASS = nullptr;
+    raw_ostream *OS = nullptr;
```
Comment (aeubanks) [inline llvm/lib/CodeGen/MachineVerifier.cpp:114]: this can be `raw_ostream &`?

**Review by aeubanks — APPROVED** [review]

File: llvm/lib/CodeGen/MachineVerifier.cpp (hunk 1 above)
Comment (arsenm) [reply llvm/lib/CodeGen/MachineVerifier.cpp:114]: Yes. Originally I was going to add the null checks like the IR verifier does, but it got messy

**Review by aeubanks — APPROVED** [review]
lgtm

**Comment by mikaelholmen** [comment]
I noticed that if I run llc now, some input that previously failed in the
```
MBB->getParent()->verify();
```
calls in ```LiveRangeCalc::findReachingDefs``` and then caused nice verifier printouts now just result in
```LLVM ERROR: Found 1 machine code errors.```
Before I got the full output like
```
[whole function]
*** Bad machine code: Reading virtual register without a def ***
- function:    main
- basic block: %bb.0  (0x5614450f2d88)
- instruction: CMP32ri8 %0:gr32, 42, implicit-def $eflags
- operand 0:   %0:gr32
LLVM ERROR: Found 1 machine code errors.
```
Is this expected? The commit message only talks about reducing output for llvm-reduce?
Can be reproduced with
```llc -O0 -mtriple i686-- -o /dev/null foo.ll -run-pass=register-coalescer```
on
```
---
name:            main
tracksRegLiveness: true
body:             |
  bb.0:
    successors:

    %0:gr32 = IMPLICIT_DEF
    %1:gr32 = COPY %0
    %1:gr32 = ADD32ri8 %1, 1, implicit-def dead $eflags
    CMP32ri8 %0, 42, implicit-def $eflags

...
```

**Comment by arsenm** [comment]
> I noticed that if I run llc now, some input that previously failed in the
> 
> ```
> MBB->getParent()->verify();

This needs to pass in the output stream to the verify call, the default is now no output 

> ```
> 
> calls in `LiveRangeCalc::findReachingDefs` and then caused nice verifier printouts now just result in `LLVM ERROR: Found 1 machine code errors.` Before I got the full output like
> 

I'm actually trying to remove all the fatal errors from LiveRangeCalc but not sure when I'll finish it

**Comment by arsenm** [comment]
> > I noticed that if I run llc now, some input that previously failed in the
> > ```
> > MBB->getParent()->verify();
> > ```
> 
> This needs to pass in the output stream to the verify call, the default is now no output
> 

Done in b7b945b09cddad128e42d82d759e74f0cd5b0ee3

**Pull request merged**